### PR TITLE
Document endpoint auth methods

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
Closes https://github.com/seamapi/docs/issues/447

### CODE REVIEW

File: src/layouts/api-endpoint.hbs [hunk 1]
```diff
@@ -4,6 +4,12 @@
 {{request.preferredMethod}} {{path}} ⇒ {{#if response.responseKey}}{{#if (eq response.responseType "resource_list")}}{ {{response.responseKey}}: [{{response.resourceType}}, …] }{{else}}{ {{response.responseKey}} }{{/if}}{{else}}void{{/if}}
 ```
 
+{{#if authMethods}}
+{% hint style="info" %}
+**Authentication Methods:** {{#each authMethods}}{{#if @index}}, {{/if}}`{{this}}`{{/each}}
```
Comment (razor-x) [inline src/layouts/api-endpoint.hbs:9]: The auth enums should map to strings like `Client Session Token` and `API Key`

**Comment by DebbieAtSeam** [comment]
@andrii-balitskyi I'm not sure about the placement and formatting of the note. I think it's important information, but is there a way to present the info. that's not in a hint box? Something like this (but prettier)?
image

File: src/layouts/api-endpoint.hbs (hunk 1 above)
Comment (andrii-balitskyi) [reply src/layouts/api-endpoint.hbs:9]: d8de1d31655849efde47784dd715a0d50b3c2f51

**Comment by andrii-balitskyi** [comment]
@DebbieAtSeam sure, but wasn't able to make it look prettier with the current setup.
Here's how the section currently looks like:
image

**Comment by DebbieAtSeam** [comment]
@andrii-balitskyi I do like that ul better than the hint box, with the exception of the capitalization. Could you please explain a bit more about why users need this info.? For example, why do only some methods accept a client session token? Also, if the ul takes up too much space, you could try it as a plain comma-separated list in a p. Thanks!

File: src/lib/layout/api-endpoint.ts [hunk 2]
```diff
@@ -43,13 +48,34 @@ export interface ApiEndpointLayoutContext {
...
 export function setEndpointLayoutContext(
   file: Partial<ApiEndpointLayoutContext>,
   endpoint: Endpoint,
 ): void {
   file.description = endpoint.description
   file.title = endpoint.title
   file.path = endpoint.path
+  file.authMethods = endpoint.authMethods.map(
+    (method) => seamAuthMethodToDisplayNameMap[method],
+  )
```
Comment (razor-x) [inline src/lib/layout/api-endpoint.ts:78]:
```suggestion
  file.authMethods = endpoint.authMethods.filter(method => method !== 'console_session_token').map(
    (method) => seamAuthMethodToDisplayNameMap[method],
  )
```

**Comment by razor-x** [comment]
@DebbieAtSeam There are past examples (which I cannot post here) where having this information would have been very helpful to address customer and internal question.

**Review by DebbieAtSeam — APPROVED** [review]

File: docs/api/acs/access_groups/add_user.md [hunk 3]
```diff
@@ -116,6 +116,12 @@ nil
 
 {% endtabs %}
 
+## Authentication Methods
+
```
Comment (DebbieAtSeam) [inline docs/api/acs/access_groups/add_user.md:120]: @andrii-balitskyi Nbd, but I prefer only initial caps, so "API _k_ey" and so on. Also, I think @razor-x mentioned to me today that Console session would become an internal-facing option. Thanks!

**Review by DebbieAtSeam — APPROVED** [review]

File: docs/api/acs/access_groups/add_user.md (hunk 3 above)
Comment (andrii-balitskyi) [reply docs/api/acs/access_groups/add_user.md:120]: Ok, I update the names here ff58ffb5677cdafe22bc9fb9fb9d850c0e9b384a.

@razor-x so, should we omit displaying console session auth method?

File: src/lib/layout/api-endpoint.ts (hunk 2 above)
Comment (andrii-balitskyi) [reply src/lib/layout/api-endpoint.ts]: https://github.com/seamapi/docs/pull/451/files#diff-af3678a98fbdb8e2301295f980f16eb094a6e203141a991b4e0b19128043e708R77-R82

**Review by razor-x — APPROVED** [review]

**Review by DebbieAtSeam — APPROVED** [review]
@andrii-balitskyi Is there an importance or other logical order to the way in which the auth methods get listed in the bulleted lists? If not, is there a way to display them in alpha order? If not, it's nbd. Thanks!

File: src/lib/layout/api-endpoint.ts [hunk 4]
```diff
@@ -51,6 +74,13 @@ export function setEndpointLayoutContext(
   file.title = endpoint.title
   file.path = endpoint.path
 
+  file.authMethods = endpoint.authMethods
+    .filter(
+      (method): method is PublicSeamAuthMethod =>
+        method in seamAuthMethodToDisplayNameMap,
+    )
+    .map((method) => seamAuthMethodToDisplayNameMap[method])
```
[the remark this answers is not in the document]
Comment (andrii-balitskyi) [reply src/lib/layout/api-endpoint.ts:82]: 0389eba7d95f06a196042d1e6b3171ad3c170b25

**Comment by DebbieAtSeam** [comment]
Thanks, @andrii-balitskyi!

**Pull request merged**